pkg/vm/stackitem/json.go

```diff
@@ -173,6 +177,11 @@ func (d *decoder) decode() (Item, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if d.count--; d.count < 0 && tok != json.Delim('}') && tok != json.Delim(']') {
```

Comment:
1. Can we separate state changing and condition checking?
2. It's `== 0` in C#, are we 100% compatible?

Reply:
I was also a bit puzzled about 2, in C# the result of post-decrement is compared, so an error occurs if `count` is 0 before the decrement.

Reply:
Looks like two years with no C programming is enough to forget about the difference between `++var` and `var++`